backend/ibutsu_server/controllers/widget_config_controller.py

```diff
@@ -23,22 +24,46 @@ def add_widget_config(widget_config=None, token_info=None, user=None):
     if not connexion.request.is_json:
         return RESPONSE_JSON_REQ
     data = connexion.request.json
+
+    # bad request checks
     if data["widget"] not in WIDGET_TYPES.keys():
         return "Bad request, widget type does not exist", HTTPStatus.BAD_REQUEST
+    # TODO: openapi plugins to support exclusive options at the schema level
+    if (
+        not any(data.get(key) for key in ["portal_id", "portal", "project_id", "project"])
+        or (data.get("project_id") or data.get("project"))
+        and (data.get("portal_id") or data.get("portal"))
+    ):
```

Comment:
Maybe serialize this via pydantic or a dataclass to ensure data consistency?

Reply:
Taken together with processing the project/portal in order to set the respective ID, I think this would combine a few logic blocks.

The model class itself may suffice, I'll investigate this further.

Thanks!